### ISSUE TITLE
Laravel livewire wire:model directive gets {value: 'value'} instead of 'value' on change

### ISSUE THREAD

**Issue by Briareos17** [jshjohnson/Choices#943]
Laravel livewire wire:model directive gets {value: 'value'} instead of 'value' on change

add wire:model to select, init choices and select something.

after selecting something it should return clear value except of json object unless it's multiple

![image](https://user-images.githubusercontent.com/2332158/113900386-6f3d0e80-97d6-11eb-8071-aa5ba09f0c72.png)
![image](https://user-images.githubusercontent.com/2332158/113900447-7e23c100-97d6-11eb-9980-9e231f7921f0.png)
![image](https://user-images.githubusercontent.com/2332158/113901334-6dc01600-97d7-11eb-9918-dffe81ff60d4.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 20H2
 - Browser Chrome
 - Version 89.0.4389.114

**Comment by Briareos17** [comment]
Any suffestions?

**Comment by ParadiseFallen** [comment]
you can use this
![image](https://user-images.githubusercontent.com/50731756/120794857-7025be80-c541-11eb-8dd2-12ef2991ce54.png)

**Comment by Briareos17** [comment]
Nope, I'm using a module with internal livewire call, so for now i catching updating hook on livewire side and processing input from choices selects.

**Comment by Briareos17** [comment]
> That's what I do too, but it seems there could be a better solution.
> 
> `if( is_array($this->customer_id) ) { // choices.js value is an array $this->customer_id = $this->customer_id['value']; }`

Made exactly same thing but this is a crutch

**Comment by flymke** [comment]
Actually that works:
`... wire:change="$set('customer_id', $event.target.value)"`

**Comment by Briareos17** [comment]
> Actually that works: `... wire:change="$set('customer_id', $event.target.value)"`

Yep, but, I have module that return me a function in x-data, there I call this.$wire.set() and so on

**Comment by NedStrk** [comment]
Having the same issue. Echoing a model's value in blade gets actual value, but if inspecting the value in the Livewire's hook `updatedFoo($value)` gets an object of `{value: $value}`.

**Comment by dawid7766** [comment]
I had similar problem and found a working solution there: https://forum.laravel-livewire.com/t/select-and-select-multi/1120

However, choices.js was losing style after update DOM, so I added to div with x-data: 
`wire:key="rand()"
wire:ignore`

**Comment by Ikozman** [comment]
maybe someone has already solved the problem with {value: 'value'}

**Comment by dawid7766** [comment]
@Ikozman That solution works for me. Inside component, I use model by entangle.

Here is code for component with Choices.js field:
```
@props(['name', 'model', 'options', 'classes' => '', 'placeholder' => '-- Select --'])

<div wire:key="{{ rand() }}" class="{{ $classes }}">
    <label>{{ $name }}</label>
    <div x-data="{ model: @entangle($model) }"
         x-init="
             choices = new Choices($refs.select, {
                noResultsText: 'No results found',
                searchPlaceholderValue: 'Search'
            });
            choices.setChoiceByValue(model + '');
            choices.passedElement.element.addEventListener('change', function(event) {
                model = event.detail.value;
            }, false);
        ">

        <select x-ref="select">
            <option value="">{{ $placeholder }}</option>

            @foreach($options as $key => $option)
                <option value="{{$key}}">{{$option}}</option>
            @endforeach
        </select>
    </div>

    @error($model)
        <p>{{ $message }}</p>
    @enderror
</div>
```
Use component with tag below:
`<x-form.select model="model.field" :options="$options" name="Field" />`

**Comment by slisarenko** [comment]
Confirming this issue still persists in 2025 with Choices.js v10.1.0 and Livewire v3.6.3
